cfme/containers/provider/openshift.py

```diff
@@ -47,3 +53,85 @@ def from_config(prov_config, prov_key):
             hostname=prov_config.get('hostname', None) or prov_config['ip_address'],
             port=prov_config['port'],
             provider_data=prov_config)
+
+    def custom_attributes(self):
+        """returns custom attributes in {name: {data}}"""
+        response = self.appliance.rest_api.get(
+            path.join(self._href(), 'custom_attributes'))
+        out = {}
+        for attr_dict in response['resources']:
+            attr = self.appliance.rest_api.get(attr_dict['href'])
+            out[attr['name']] = attr
+        return out
+
+    def add_custom_attributes(self, names, values, field_types):
```

Comment:
As there is a 1:1 mapping for this, doesn't it make more sense to create maybe a NamedTuple, or even a class for a Custom Attribute and pass in a list of those objects, rather than breaking them apart?